File: 12-BIT-MANIPULATION/Go/number-of-1-bits.go
package main

// https://leetcode.com/problems/number-of-1-bits/

func hammingWeight(n int) int {
    c := 0
    for n > 0 {
        if n & 1 == 1 {
            c++
        }
        n >>= 1
    }
    return c
}

File: 12-BIT-MANIPULATION/PHP/number-of-1-bits.php
<?php

// https://leetcode.com/problems/number-of-1-bits/

class Solution
{
    function hammingWeight(int $n): int
    {
        $c = 0;
        while ($n > -0) {
            if ($n & 1) {
                $c++;
            }

            $n >>= 1;
        }

        return $c;
    }
}